### PR TITLE
DAOS-15969 : Disable downstream testing on release/2.4

### PR DESCRIPTION
**Pull request by grom72**
Remove support for the 2.4 release from all jobs.

### CODE REVIEW

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
We are looking for the minimum (not maximum) amount of change required to simply stop testing with `release/2.4`.  There is no need to remove support for it from ancillary, test and support functions.

File: vars/distroVersion.groovy
[the remark this answers is not in the document]
Comment (grom72) [reply vars/distroVersion.groovy:1]: But do we need them after the release 2.4 removal?

File: Jenkinsfile [hunk 1]
```diff
@@ -148,15 +148,12 @@ pipeline {
                     steps {
                         script {
                             assert(daosLatestVersion('master', 'el8').matches(/2.7\.\d+.*/))
-                            assert(daosLatestVersion('release/2.4', 'el8').matches(/2.[34]\.\d+.*/))
```
[the remark this answers is not in the document]
Comment (grom72) [reply Jenkinsfile:151]: Won't it become an empty code as we will not use the pipeline with the release 2.4?

[the remark this answers is not in the document]
Comment (brianjmurrell) [reply Jenkinsfile:151]: No.  This is a test case.

File: vars/distroVersion.groovy
[the remark this answers is not in the document]
Comment (brianjmurrell) [reply vars/distroVersion.groovy:1]: They are needed to satisfy the requirement that we can use `pipeline-lib` to re-build older releases when necessary, for support purposes, etc.

[the remark this answers is not in the document]
Comment (grom72) [reply vars/distroVersion.groovy:1]: Done

File: vars/branchTypeRE.groovy
[the remark this answers is not in the document]
Comment (grom72) [reply vars/branchTypeRE.groovy:1]: Done

File: Jenkinsfile [hunk 2]
```diff
@@ -148,15 +148,12 @@ pipeline {
...
                             assert(daosLatestVersion('release/2.6', 'el8').matches(/2.[56]\.\d+.*/))
                         }
                     }
                 }
                 stage('distroVersion() tests') {
                     steps {
-                        distro_version_test('release/2.4', 'el8', '8')
-                        distro_version_test('release/2.4', 'leap15', '15')
```
[the remark this answers is not in the document]
Comment (grom72) [reply Jenkinsfile]: Done

File: Jenkinsfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (grom72) [reply Jenkinsfile]: Done

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**